### ISSUE TITLE
WiFi is more inconsistent/unstable in Dopamine than Fugu15 Max

### ISSUE THREAD

**Issue by NotDarkn** [opa334/Dopamine#134]
**Device - iOS - Version:**
iPhone 11 Pro - 15.1 - v1.0.3

## tl;dr
In the beta versions of Dopamine, formally known as Fugu15 Max, almost all of the versions (b8 is a bit of an exception but was fine) had very consistent Wi-Fi stability and Wi-Fi would work almost all of the time after jailbreaking. 

However when I went to jailbreak with Dopamine, the Wi-Fi stability was... horrendous to say so myself. The majority of the sessions where I had successfully jailbroken would turn into me rebooting because Wi-Fi was broken, and I'd have to keep rebooting my device until I got a successful Wi-Fi session. This has been the case for the past 3 versions of Dopamine and the worst part about it is that it's very inconsistent. Sometimes it'd take me 1 or 2 reboots for Wi-Fi to work, some others it'd take 10+ reboots.

I'm not sure if I'm just unlucky, have an unreliable device, or the jailbreak has gotten worse, but this has been occurring for me for the past 3 versions of Dopamine so I felt like I'd make an issue on it.

#### What is Wi-Fi stability?
_To further clarify on what I mean by "Wi-Fi stability", basically when you jailbreak with Dopamine or Fugu15 Max there is a chance that your Wi-Fi ends up not working. This then requires the user to reboot and then re-jailbreak until you get a session with working Wi-Fi._

**Comment by opa334** [comment]
There is no difference between Dopamine 1.0.3 and Fugu15 Max that could affect any of this. Also wifi has always worked fine on all of my devices.

**Comment by opa334** [comment]
Check if you got any odd looking crash logs from any processes and if so, send them here

**Comment by augustobob** [comment]
In my experience with fugu15 on 15.1.1 12pm since first beta my wifi became unstable and sometimes takes a long time to connect after a reboot or jailbreak, unless I connect manually then it worked fine.
The problem persisted after removing fugu15 and in a research for the problem I messed some files on /something/preboot/something/caches which affected my wifi too, but no solution. 
I can’t confirm on dopamine since I bootlooped hours before the first dopamine release and I’m jailed now on iOS 16.
Ik it can’t help, but it was my experience

**Comment by opa334** [comment]
> > Check if you got any odd looking crash logs from any processes and if so, send them here
> 
> 
> 
> How would you like me to do this? Through Filza, through KrashKop, through both?

Both are the same

**Comment by opa334** [comment]
Yeah these SpringBoard crashes are not the reason for any wifi issues

**Comment by opa334** [comment]
Send one of each from the following:
heard
kbd
mediaserverd
ReportMemoryException
whoami

**Comment by Cryptiiiic** [comment]
None of these make sense tf

**Comment by augustobob** [comment]
What’s the difference between Xina and dopamine exploit? Why Xina doesn’t have this wifi bug present in fugu/dopamine?

**Comment by augustobob** [comment]
> > What’s the difference between Xina and dopamine exploit? Why Xina doesn’t have this wifi bug present in fugu/dopamine?
> 
> The main difference between the two exploits is the versions they support. The exploit that XinaA15 uses supports iOS 15.0-15.1.1 only, while Fugu15’s exploit supports iOS 15.0-15.4.1 only.
> 
> I don’t know the technical reasons on why Fugu15’s exploit has a Wi-Fi bug, however all I can say is: The exploit that Fugu15 uses (and also Dopamine) has a Wi-Fi bug which can be fixed but hasn’t fully been fixed yet. The only workaround for this is manually or automatically disabling Wi-Fi for affected devices on iOS 15.0-15.3.1.

People’s comments over there lead me to think both used the same exploit, thanks for your clarification

**Comment by Midnight145** [comment]
I am said "other person"

We both use ExpressVPN, unsure if that specifically matters.
You just click on this toggle after jailbreaking, and wifi/cellular works as normal again.
![image](https://github.com/opa334/Dopamine/assets/23142346/b1511315-809e-4ed3-8a23-cd3710c09376)
Userspace reboots definitely seem to be what triggers it, as I can trigger it manually via either the "reboot device" postscript in Sileo or the reboot userspace button in Dopamine itself.

**Comment by opa334** [comment]
Yeah so might be a stock bug.

I have Mullvad installed on my device and no such issue, so not sure.

**Comment by opa334** [comment]
stock bug (most likely)
